src/Test.UnitTests.Sarif.Driver.Net48/AnalyzeTestCommand.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Reflection;
+using Microsoft.CodeAnalysis.Sarif.Driver;
+
+namespace Test.UnitTests.Sarif.Driver.Net48
+{
+    public class AnalyzeTestCommand : MultithreadedAnalyzeCommandBase<AnalyzeTestContext, AnalyzeTestOptions>
```

Comment:
>AnalyzeTestCommand [](http://example.com/codeflow?start=17&length=18)

why isn't our standard testing catching this problem? it seems like configuring some tests to run in downlevel CLR would be preferable than adding a specialized test assembly.

what is your thinking here?

Reply:
The normal testing runs on .NET 6 and Path.GetExtension silently ignores illegal characters in later versions.  It only throws the exception in older versions of .NET.

Reply:
This is good information but doesn't describe your rationale for creating a new test binary for this specific case. Maybe you didn't want to take on the performance costs for running tests on multiple frameworks?

I think those costs might be worth taking on. The SARIF binary in particular has a lot of fairly low level API (mostly in the IO namespace, the URI handling, etc.) where there are some troublesome compat issues like this one.

So I suggest you add net482 to the target frameworks for the core unit test dll and move the new tests over to it.

<TargetFrameworks>net6.0;</TargetFrameworks>

Reply:
Attempting to build Test.UnitTests.Sarif.Driver against net48 results in several instances of `error CS8831: Target runtime doesn't support covariant types in overrides.'